File: EPSILON/SAMPLE FILE/E-publishing/Tools/Altova/Altova/XMLSpy2006/spl/java/altova/text/tablelike/RecordBasedParser.java
[call GenerateFileHeader("RecordBasedParser.java")]
package com.altova.text.tablelike;

public abstract class RecordBasedParser {
    private IRecordBasedParserObserver m_Observer = null;

    public IRecordBasedParserObserver getObserver() {
        return m_Observer;
    }

    public void setObserver(IRecordBasedParserObserver rhs) {
        m_Observer = rhs;
    }

    public abstract int parse(String buffer) throws MappingException;

    protected void notifyAboutRecordFound(String\[\] fields) {
        if (null != m_Observer)
            m_Observer.notifyAboutRecordFound(new Record(fields));
    }

    protected void notifyAboutRecordFound(StringList fields) {
        if (null != m_Observer)
            m_Observer.notifyAboutRecordFound(new Record(fields));
    }
}

File: EPSILON/SAMPLE FILE/E-publishing/Tools/Altova/Altova/XMLSpy2006/spl/cpp/AltovaText/RecordBasedParser.cpp
[call GenerateFileHeader("RecordBasedParser.cpp")]
#include "stdafx.h"
#include "RecordBasedParser.h"


namespace altova
{
namespace text
{
namespace tablelike
{


CRecordBasedParser::CRecordBasedParser()
	: m_pObserver(NULL)
{}

CRecordBasedParser::~CRecordBasedParser() {} // NO-OP

IRecordBasedParserObserver* CRecordBasedParser::GetObserver() const
{
	return m_pObserver;
}
void CRecordBasedParser::SetObserver(IRecordBasedParserObserver* rhs)
{
	m_pObserver= rhs;
}
void CRecordBasedParser::NotifyObserverAboutRecordFound(const TStringList& fields)
{
	if (NULL!=m_pObserver) m_pObserver->NotifyAboutRecordFound(fields);
}

CMappingException::CMappingException(const tstring& message)
:	m_Message(message)
{}
CMappingException::CMappingException()
:	m_Message(_T(""))
{}
const tstring& CMappingException::GetMessage() const
{
	return m_Message;
}
void CMappingException::SetMessage(const tstring& message)
{
	m_Message= message;
}

} // namespace tablelike
} // namespace text
} // namespace altova
